src/D2L.CodeStyle.TestAnalyzers/TestCaseData/TestCaseDataAnalyzer.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace D2L.CodeStyle.TestAnalyzers.TestCaseData {
+	[DiagnosticAnalyzer( LanguageNames.CSharp )]
+	internal sealed class TestCaseDataAnalyzer : DiagnosticAnalyzer {
+		public const string DiagnosticId = "D2L0010";
+		private const string Category = "Safety";
+
+		private const string Title = "Ensure test does not contain named property 'Throws' in TestCaseData.";
+		private const string Description = "Named property 'Throws' should not be used in TestCaseData.";
+		internal const string MessageFormat = "Use Assert.Throws or Assert.That in your test case instead of 'Throws' in TestCaseData for NUnit 3 compatibility";
+
+		private static readonly DiagnosticDescriptor Rule = new DiagnosticDescriptor(
+			DiagnosticId,
+			Title,
+			MessageFormat,
+			Category,
+			DiagnosticSeverity.Warning,
+			isEnabledByDefault: true,
+			description: Description
+		);
+
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create( Rule );
+
+		public override void Initialize( AnalysisContext context ) {
+			context.RegisterSyntaxNodeAction(
+				AnalyzeSyntaxNode,
+				SyntaxKind.InvocationExpression
+			);
+		}
+
+		private void AnalyzeSyntaxNode( SyntaxNodeAnalysisContext context ) {

```

Comment:
I think this method could be better written like this:

```csharp
private void AnalyzeSyntaxNode( SyntaxNodeAnalysisContext context ) {
	var root = context.Node as InvocationExpressionSyntax;
	if( root == null ) {
		return;
	}

	var symbol = context.SemanticModel.GetSymbolInfo( root );
	if( symbol.Symbol == null ) {
		return;
	}

	var method = symbol.Symbol as IMethodSymbol;
	if( method == null ) {
		return;
	}

	if( method.Name == "Throws" && method.ContainingType.Name == "TestCaseData" ) {
		var diagnostic = Diagnostic.Create( Rule, root.GetLocation() );
		context.ReportDiagnostic( diagnostic );
	}
}
```

This handles all cases of a method invocation expression and doesn't care so much about how the syntax is laid out. I tested it and it works locally, with some changes to the tests (to add a reference to NUnit and make the tes code actually compile). What do you think?